src/main/kotlin/org/rust/ide/typing/RsPairedBraceMatcher.kt

```diff
@@ -0,0 +1,80 @@
+package org.rust.ide.typing
+
+import com.intellij.codeInsight.highlighting.PairedBraceMatcherAdapter
+import com.intellij.openapi.editor.highlighter.HighlighterIterator
+import com.intellij.openapi.fileTypes.FileType
+import com.intellij.psi.tree.TokenSet
+import org.rust.lang.RsFileType
+import org.rust.lang.RsLanguage
+import org.rust.lang.core.psi.RsTokenElementTypes
+import org.rust.lang.core.psi.RsTokenElementTypes.*
+
+class RsPairedBraceMatcher : PairedBraceMatcherAdapter(RsBraceMatcher(), RsLanguage) {
+
+    private object Holder {
+        val TYPE_TOKENS = TokenSet.orSet(COMMENTS_OR_WHITE_SPACE_TOKEN_SET,
+            TokenSet.create(
+                RsTokenElementTypes.IDENTIFIER,
```

Comment:
There will be a lot of tokens here, so it make sense to import `RsTokenElemntTypes.*`